server/selection.go

```diff
@@ -9,6 +18,161 @@ type BroadcastSessionsSelector interface {
 	Clear()
 }
 
+type sessHeap []*BroadcastSession
+
+func (h sessHeap) Len() int {
+	return len(h)
+}
+
+func (h sessHeap) Less(i, j int) bool {
+	return h[i].LatencyScore < h[j].LatencyScore
+}
+
+func (h sessHeap) Swap(i, j int) {
+	h[i], h[j] = h[j], h[i]
+}
+
+func (h *sessHeap) Push(x interface{}) {
+	sess := x.(*BroadcastSession)
+	*h = append(*h, sess)
+}
+
+func (h *sessHeap) Pop() interface{} {
+	old := *h
+	n := len(old)
+	sess := old[n-1]
+	old[n-1] = nil
+	*h = old[:n-1]
+
+	return sess
+}
+
+func (h *sessHeap) Peek() interface{} {
+	if h.Len() == 0 {
+		return nil
+	}
+
+	return (*h)[0]
```

Comment:
We peek from the front since the minimum element is at the 0th index (see the [container/heap](https://golang.org/pkg/container/heap/) docs). We pop from the back since since [heap.Pop()](https://golang.org/src/container/heap/heap.go?s=2190:2223#L50) swaps the 0th element with the last element before calling the interface's `Pop()` method.

Reply:
Maybe add a comment to the code explaining this?

Reply:
Added a comment in 5bd2145